foreshadow/transformers/smart.py

```diff
@@ -52,17 +52,31 @@ class Encoder(SmartTransformer):
     """Automatically Encodes Categorical Features
 
     If there are less than 30 categories, then OneHotEncoder is used, if there are more
-    then HashingEncoder is used. If used in a y_var context, LabelEncoder is used.
+    then HashingEncoder is used. If the columns containing a delimmeter exceed delim_cuttoff then a
+    DummyEncoder is used (set cutoff to -1 to force). If used in a y_var context, LabelEncoder is used.
 
     """
 
-    def _get_transformer(self, X, y=None, unique_num_cutoff=30, **fit_params):
+    def _get_transformer(
+        self, X, y=None, unique_num_cutoff=30, delim_cutoff=0.5, **fit_params
+    ):
         data = X.iloc[:, 0]
         col_name = X.columns[0]
         unique_count = len(data.value_counts())
+
+        delimeters = [",", ";", "\t"]
+        delim_count = [
+            data.astype("str").str.count(d).fillna(0).astype(bool).sum(axis=0)
+            for d in delimeters
+        ]
+        delim_ratio = np.sum(delim_count) / data.count()
```

Comment:
No sure why we use the sum to compare to the cut-off while later we only select one delimiters. Normally, only one delimiter will be used.

Reply:
Will be resolved as a result of changing the way the dummy encoder is selected.